### PR TITLE
Enable additional linters

### PR DESCRIPTION
**Pull request by VenelinMartinov**
This enables some additional linters for the bridge codebase. I've mostly taken the linters pu/pu uses from https://github.com/pulumi/pulumi/pull/17456

The ones I've kept off are:
- `exhaustive` - checks all switch statements are exhaustive. We seem to use switch statements in many places to check if a value is in a given list. This pattern conflicts with the linter, I've opted to keep it off.
- `perfsprint` - checks for commonly misused Printf statments - We use fmt.Sprintf("string") without formatters, fmt.Error("string") without formatters and a few other similar patterns. I've kept it off since it'd make this change a few times larger, seems unnecessary for now.
- `depguard` - pu/pu uses this to enforce the new major version of the protobuf library - We have not yet upgraded and the upgrade is non-trivial as there were some breaking changes.
- `goheader` - pu/pu uses this to enforce copyright headers - We lack these in quite a few places. Enforcing this would make the change a lot larger and it's not clear that it is necessary.
- `prealloc` - this flags up arrays which are not pre-allocated - The linter, however, will also flag up cases which can not be pre-allocated, for example if there is an early return in a loop. Opted to keep it off.

The changes are:
 - [b44efa3](https://github.com/pulumi/pulumi-terraform-bridge/pull/2559/commits/b44efa3b0e4c9cfedd470adc0b833713f082b6b4) and [915aa1b](https://github.com/pulumi/pulumi-terraform-bridge/pull/2559/commits/915aa1b09bcebc6c62cae7432ea2738577c07835) - all the automated fixes
 - [a8bb128](https://github.com/pulumi/pulumi-terraform-bridge/pull/2559/commits/a8bb1281a598ffe335ed8d5620d08f1cbf4bd8db) additional manual fixes

### CODE REVIEW

**Review by t0yv0 — APPROVED** [review]

File: pkg/tfbridge/schema.go [hunk 1]
```diff
@@ -1076,7 +1075,7 @@ func MakeTerraformOutputs(
 
 		// Next perform a translation of the value accordingly.
 		out := MakeTerraformOutput(ctx, p, value, tfi, psi, assets, supportsSecrets)
-		//if !out.IsNull() {
+		// if !out.IsNull() {
```
Comment (lunaris) [inline pkg/tfbridge/schema.go:1078]: Maybe an opportunity to delete this? Looks like dead code?

File: pkg/tfgen/parse_markdown.go [hunk 2]
```diff
@@ -271,7 +272,7 @@ func parseParameterSection(node *bf.Node, consumeNode func(node *bf.Node)) (para
 				return -1, nil, nil, err
 			}
 
-			if ps == nil || node == nil || node.Next == nil {
+			if ps == nil || node.Next == nil {
```
Comment (lunaris) [inline pkg/tfgen/parse_markdown.go:275]: This is safe because we check `node != nil` up above, I'm guessing?

**Review by lunaris — APPROVED** [review]
Not sure my approval is sufficient alone but LGTM :boom:

File: pkg/tfgen/parse_markdown.go (hunk 2 above)
Comment (VenelinMartinov) [reply pkg/tfgen/parse_markdown.go:275]: Yeah, exactly - it was flagged up by the linter

File: pkg/tfbridge/schema.go (hunk 1 above)
Comment (VenelinMartinov) [reply pkg/tfbridge/schema.go]: Yeah, good point, thanks for flagging it.

**Review by iwahbe — APPROVED** [review]
LGTM!

**Pull request merged**